conf.d/55canvas-install

```diff
@@ -32,27 +32,35 @@ export BUNDLE_PATH="vendor/bundle"
 bundle config set without 'mysql sqlite'
 bundle install
 
+npm install -g yarn@1.22.11 wsrun
+yarn config set strict-ssl false
 mkdir -p $YARN_CACHE
-yarn install ---scripts-prepend-node-path --network-concurrency 1 --cache-folder $YARN_CACHE
+yarn install --scripts-prepend-node-path --cache-folder $YARN_CACHE
 
-CANVAS_BUILD_CONCURRENCY=1 bundle exec rake canvas:compile_assets
+# COMPILE_ASSETS_BRANC_CONFIGS variable:
+# https://github.com/instructure/canvas-lms/issues/2023
 
-# apply patch to work around install issue; see
-# https://groups.google.com/g/canvas-lms-users/c/A5l8B84SezM
-_patch=/usr/local/src/app_models_course.rb.patch
-git apply $_patch
-rm -rf $_patch
+# fix for webpack failure
+# https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported
+sed -i 's/5120/5120 --openssl-legacy-provider/g' package.json
```

Comment:
I'm a little skeptical of this fix tbh. Afaik `--openssl-legacy-provider` relates to this "https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html" which is required because an issue was found with NPM's default config prior to Node 17. Meaning that it's default configuration is now unsupported (and possibly insecure). 

I'm unsure if it's viable or not but this answer on the same issue lists 2 potential alternative fixes https://stackoverflow.com/a/73027407

Reply:
thanks man, i must have missed/skimmed over that. this issue is a little hairy. the problem isn't the config but old packages which use TLS from my understanding

- the suggested npm fix breaks the dependency graph by bumping major semver versions of packages which with canvas is a very big deal. and it still fails to resolve after that. yarn is used which npm conflicts with. i experimented with different approaches a bit but no positive outcome
- the suggested webpack fix simply doesn't fix the issue. i confirmed that it takes effect in that if an invalid value is used on purpose, webpack will fail because of unknown hash function. however if it's changed to `xxhash64` or `md5` as advised, the same issue still comes up

this is node v18 (latest LTS). i think waiting until upstream updates package.json and having the hack in the meantime is currently our only sane way out